elements/rh-footer/rh-footer.css

```diff
@@ -0,0 +1,327 @@
+}
+
+* {
+  box-sizing: border-box;
+}
+
+/**
+ * Debugging
+ */
+:host([debug]) *:not(.spacer, .base, a, svg) {
+  position: relative;
+  outline: 2px dotted red;
+  /* make sure we have some */
+  min-height: 25px;
+}
+
+:host([debug]) *:not(.spacer, .base, a, svg)::after {
+  content: attr(part);
+  display: block;
+  position: absolute;
+  color: white;
+  background-color: darkslategray;
+  padding: 0.3em;
+  top: 0;
+  right: 0;
+  font-size: 0.7rem;
+}
+
+/**
+ * Regions
+ */
+.section {
+  padding: var(--pf-global--spacer--xl, 32px) var(--_section-side-gap);
+}
+
+.header {
+  background-color: #212427;
+  /* children should flex wrap on mobile */
+  display: flex;
+  flex-wrap: wrap;
+  gap: var(--pf-global--spacer--xl, 32px);
+  align-items: center;
+  position: relative;
+}
+
+.header::after {
+  display: block;
+  content: '';
+  background-color: var(--_border-color);
+  height: 1px;
+  position: absolute;
+  bottom: 0px;
+  width: calc(100% - var(--_section-side-gap) * 2);
+  left: var(--_section-side-gap);
+}
+
+.header-primary {
+  flex: 1 1 auto;
+}
+
+.header-secondary {
+  /* secondary should be push to the end */
+  /* of the line on mobile */
+  flex: 0 1 auto;
+}
+
+.main {
+  background-color: #212427;
+  display: grid;
+  gap: var(--pf-global--spacer--xl, 32px);
+}
+
+.footer {
+  --link-font-size: 12px;
+  /* reduce the line-height for footer links */
+  line-height: 100%;
+  background-color: #151515;
+  display: grid;
+  grid-template-columns: 1fr;
+  grid-template-areas:
+    'logo'
+    'primary'
+    'spacer'
+    'secondary'
+    'tertiary';
+  gap: 32px 24px;
+}
+
+.footer-logo {
+  grid-area: logo;
+  width: 42px;
+}
+
+.footer-logo-image {
+  fill: var(--_icon-color);
+}
+
+.footer-logo-image:is(:hover, :focus-within) {
+  fill: var(--_icon-color-hover);
+}
+
+.footer-primary {
+  grid-area: primary;
+}
+
+.footer-secondary {
+  grid-area: secondary;
+  color: #d2d2d2;
+  font-size: 12px;
+  display: flex;
+  flex-direction: column;
+  gap: 16px;
+}
+
+.footer-tertiary {
+  grid-area: tertiary;
+  display: grid;
+  justify-content: flex-start;
+  align-items: center;
+}
+
+.footer-links-primary {
+  display: flex;
+  flex-direction: column;
+  gap: 8px 24px;
+}
+
+.footer-links-secondary {
+  display: flex;
+  flex-direction: column;
+  gap: 8px 24px;
+}
+
+/**
+ * Content
+ */
+.logo {
+  /* fix wierd problem where there is extra space below logo */
+  line-height: 0px;
+}
+
+.logo slot::slotted(a),
+.logo a {
+  display: inline-flex;
+  width: 156px;
+}
+
+.social-links {
+  display: flex;
+  margin-left: 0;
+  padding-left: 0;
+}
+
+.social-links rh-footer-links,
+.social-links slot::slotted(rh-footer-links) {
+  display: flex;
+  flex-direction: row;
+  gap: 16px;
+}
+
+:host(:not([is-mobile])) .links {
+  display: grid;
+  grid-template-columns: repeat(1fr, 25%);
+  grid-template-rows: repeat(2, min-content auto);
+  grid-template-columns: 25%;
+  gap: var(--rh-footer-links-column-gap, 32px);
+}
```

Comment:
`grid-template-columns` is defined twice, the second definition keeps the .links columns from taking up more then 25% when there might only be 2 sets of links instead of 4.  @coreyvickery brought this up in chat, this is my suggestion for a fix.

```Suggestion
:host(:not([is-mobile])) .links {
  display: grid;
  grid-template-columns: repeat(1fr, 25%);
  grid-template-rows: repeat(2, min-content auto);
  gap: var(--rh-footer-links-column-gap, 32px);
}
```

Reply:
Example screenshot.

Got approval that version B is how we want it to look.

![Footer content flow examples](https://user-images.githubusercontent.com/45767595/168385791-64f6b14a-8301-4f09-b49a-e02a2d0e533d.png)

Reply:
<img width="1142" alt="Screen Shot 2022-05-15 at 11 13 44" src="https://user-images.githubusercontent.com/1466420/168463565-83bf1024-3420-4627-be38-cbdec1c14f71.png">
Added 2- and 3-column layouts to the demo to confirm this fix